multicluster/controllers/multicluster/stale_controller.go

```diff
@@ -416,9 +405,15 @@ func (c *StaleResCleanupController) Run(stopCh <-chan struct{}) {
 	defer klog.InfoS("Shutting down StaleResCleanupController")
 
 	ctx, _ := wait.ContextForChannel(stopCh)
-	if err := c.RunOnce(ctx); err != nil {
-		c.Enqueue()
-		go wait.UntilWithContext(ctx, c.runWorker, time.Second)
+
+	switch c.clusterRole {
+	case LeaderCluster:
```

Comment:
We do not need "c.RunOnce(ctx)" to clean up other stale resources in the leader?

Reply:
`c.RunOnce` will call `c.cleanup`, c.cleanup will run different clean up functions based on the `c.clusterRole`. For leader, it actually will run `cleanupMemberClusterAnnounces` only. Most clean up functions in stale_controller is mainly for the member cluster.

Reply:
Yeah, it's already changed to call `cleanupMemberClusterAnnounces` only here for the leader cluster.
I think we should keep it running periodically (12 hours for now) to make sure that stale MemberClusterAnnounce can always be cleaned up. Because there will be a case that the member cluster is deleted by 'kubectl delete -f', then the MemberClusterAnnounce will become stale after 24 hours. So the `cleanupMemberClusterAnnounces` should be called periodically. 

c.RunOnce(ctx) is now called on the member cluster controller only.